### PR TITLE
Pf/gl 8

### PR DESCRIPTION
**Pull request by pford68**

### CODE REVIEW

File: src/main/java/com/giantmachines/biblio/model/Review.java [hunk 1]
```diff
@@ -20,17 +20,36 @@
...
 
     @PrePersist
-    private void addReviewTime(){
-        this.reviewTime = new Date().getTime();
+    private void setTimeCreated(){
+        this.timeCreated = this.timeCreated == null ? new Date().getTime() : this.timeCreated;
+        this.timeUpdated = new Date().getTime();
     }
 
     public Review(User reviewer, int value, String comments) {
```
Comment (ecumali) [inline src/main/java/com/giantmachines/biblio/model/Review.java]: Lombok for all the boilerplate code here (and elsewhere)

File: src/main/java/com/giantmachines/biblio/exceptions/BookUnavilableException.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.giantmachines.biblio.exceptions;
+
+import com.giantmachines.biblio.model.Book;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value = HttpStatus.BAD_REQUEST)
+public class BookUnavilableException extends Exception{
```
[the remark this answers is not in the document]
Comment (pford68) [reply src/main/java/com/giantmachines/biblio/exceptions/BookUnavilableException.java]: Yes, typo

File: src/main/java/com/giantmachines/biblio/controller/BookController.java [hunk 3]
```diff
@@ -41,6 +52,12 @@ public ResponseEntity getById(@PathVariable long id) {
         return this.buildOkResponse(new BookDetailsDto(this.service.getById(id)));
     }
 
+    @RequestMapping(value = "/{id}/reviews", method = RequestMethod.GET)
```
[the remark this answers is not in the document]
Comment (pford68) [reply src/main/java/com/giantmachines/biblio/controller/BookController.java]: OK, fair enough

File: src/main/java/com/giantmachines/biblio/model/Book.java [hunk 4]
```diff
@@ -13,8 +13,8 @@
     @ManyToOne(fetch = FetchType.EAGER)
```
[the remark this answers is not in the document]
Comment (pford68) [reply src/main/java/com/giantmachines/biblio/model/Book.java]: We do not need it. Will remove.

**Pull request merged**